File: parish360-core/src/main/java/org/parish360/core/ceremonies/dto/ChurchInfo.java
package org.parish360.core.ceremonies.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.parish360.core.dao.entities.common.Address;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ChurchInfo {
    private String familyCode;
    private String church;
    private String diocese;
    private Address churchLocation;
}

File: parish360_mobile/lib/features/ceremonies/domain/entities/church_info.dart
import 'package:parish360_mobile/core/common/entities/place.dart';

class ChurchInfo {
  final String? familyCode;
  final String? church;
  final String? diocese;
  final Place? churchLocation;

  ChurchInfo({this.familyCode, this.church, this.diocese, this.churchLocation});

  factory ChurchInfo.fromJson(Map<String, dynamic> json) {
    return ChurchInfo(
      familyCode: json['family_code'] as String?,
      church: json['church'] as String?,
      diocese: json['church'] as String?,
      churchLocation: json['church_location'] != null
          ? Place.fromJson(json['church_location'] as Map<String, dynamic>)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'family_code': familyCode,
      'church': church,
      'diocese': diocese,
      'church_location': churchLocation?.toJson(),
    };
  }
}
